app/src/main/java/com/example/bigowlapp/utils/SupervisorSmsListener.java

```diff
@@ -0,0 +1,67 @@
+package com.example.bigowlapp.utils;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.telephony.SmsMessage;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Schedule;
+import com.google.firebase.Timestamp;
+
+import java.util.Calendar;
+
+public class SupervisorSmsListener extends BroadcastReceiver {
+
+    private static final String SMS_RECEIVED = "android.provider.Telephony.SMS_RECEIVED";
+    private static final int THIRTY_MINUTES = 1800;
+    private String scheduleId;
+
+    IsSmsSenderACurrentEventSupervisor isSmsSenderACurrentEventSupervisor = new IsSmsSenderACurrentEventSupervisor();
+
+    public void onReceive(Context context, Intent receiverIntent) {
+        if (receiverIntent.getAction().equals(SMS_RECEIVED)) {
+            Bundle bundle = receiverIntent.getExtras();
+            if (bundle != null) {
+                // get sms objects
+                Object[] pdus = (Object[]) bundle.get("pdus");
+                if (pdus.length == 0) {
+                    return;
+                }
+                // large message might be broken into many
+                SmsMessage[] messages = new SmsMessage[pdus.length];
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < pdus.length; i++) {
+                    messages[i] = SmsMessage.createFromPdu((byte[]) pdus[i]);
+                    sb.append(messages[i].getMessageBody());
+                }
+                String smsSender = messages[0].getOriginatingAddress().substring(2);
```

Comment:
Cause of Case 3.

When getting the SMS sender, in my case, the sender's phone number did not contain a "+". Because our database enforces a specific format which require a "+" at the beginning of a number, the phone number received from this variable would never match the phone number from the database even if the supervisor is actually sending the SMS.

The solution is simply using the phone number formatter like the suggested solution for Case 1.

Reply:
Yes, I did that also as in your Case 1 review